From a24a6cff0e4af5418edd129767b9aa4b41c3fd98 Mon Sep 17 00:00:00 2001
From: icg_intg <icg_intg@intel.com>
Date: Mon, 18 Nov 2019 10:40:14 +0800
Subject: [PATCH 2358/2367] isys & psys package lib2600b0 for commit-id e07fe3f

FW_Manifest_ID: cd9988f344844f2cb93d98e53402892cdabac0de

FW_Build_Branch: stable_iot_gp1_ivi_20190410

FW_Build_Topic: auto_merge_queue

library version: commit-id e07fe3f

FW Bundle patch number of this release :  444266

FW tag : scci_stable_iot_gp1_ivi_20190410_20191116_0100

Change-Id: I341ab927b5e384fa307188f5bd189a41b40baf9b
Depends-on: I2060fcd20771ba6a22c7b62470cdddaad68791e1
Depends-on: Iaf2d74f66708cac4cc1051d6b8499cdeef55e860
Tracked-On: #PKT-2935
Signed-off-by: icg_intg <icg_intg@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
Signed-off-by: Ong Yang Yang <yang.yangx.ong@intel.com>
---
 drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h      | 2 +-
 .../pci/intel/ipu4/ipu4-css/lib2600/config/isys/subsystem_bxtB0.mk | 7 ++++++-
 drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/isysapi.mk   | 2 ++
 .../lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h   | 2 +-
 4 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h
index a89c17f..37a8016 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20191003
+#define IA_CSS_FW_PKG_RELEASE  0x20191115
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/config/isys/subsystem_bxtB0.mk b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/config/isys/subsystem_bxtB0.mk
index c4549f3..0bc42f9 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/config/isys/subsystem_bxtB0.mk
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/config/isys/subsystem_bxtB0.mk
@@ -49,6 +49,11 @@ WA_HSD1805168877		 = 0
 # WA to match the NofSOF & NofEOF when resuming a stream to prevent fps drop
 WA_HSD1507026600         = 1
 
+# WA related to HSD1507026600 which disables prequeuing commands to IBufCtrl (i.e. only one capture command
+# queued to HW at a time). This prevents a rare hang that can occur and is essentially a
+# better solution than the "5 extra lines" workaround that was used to fix HW bug referenced in HSD1303898875
+WA_DISABLE_IBUF_PREQUEUE = 1
+
 # Support IBUF soft-reset at stream start
 SOFT_RESET_IBUF_STREAM_START_SUPPORT = 1
 
@@ -58,6 +63,6 @@ SOFT_RESET_IBUF_STREAM_START_SUPPORT = 1
 
 # TODO: This define should be entirely removed.
 # Used in mipi_capture
-ISYS_DISABLE_VERIFY_RECEIVED_SOF_EOF     = 0
+ISYS_DISABLE_VERIFY_RECEIVED_SOF_EOF     = 1
 
 ISYS_ACCESS_BLOCKER_VERSION      = v1
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/isysapi.mk b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/isysapi.mk
index 0d06298..0911959 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/isysapi.mk
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/isysapi.mk
@@ -62,8 +62,10 @@ ISYSAPI_FW_CPPFLAGS += -I$(HIVESDK)/include/ipu/dai
 ISYSAPI_FW_CPPFLAGS += -I$(HIVESDK)/include/ipu
 
 ISYSAPI_FW_CPPFLAGS += -DWA_HSD1805168877=$(WA_HSD1805168877)
+ISYSAPI_FW_CPPFLAGS += -DWA_DISABLE_IBUF_PREQUEUE=$(WA_DISABLE_IBUF_PREQUEUE)
 
 ISYSAPI_HOST_CPPFLAGS += -DREGMEM_OFFSET=$(REGMEM_OFFSET)
+ISYSAPI_HOST_CPPFLAGS += -DWA_DISABLE_IBUF_PREQUEUE=$(WA_DISABLE_IBUF_PREQUEUE)
 
 ifeq ($(ISYS_HAS_DUAL_CMD_CTX_SUPPORT), 1)
 ISYSAPI_HOST_CPPFLAGS += -DHAS_DUAL_CMD_CTX_SUPPORT=$(ISYS_HAS_DUAL_CMD_CTX_SUPPORT)
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h
index a89c17f..37a8016 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20191003
+#define IA_CSS_FW_PKG_RELEASE  0x20191115
-- 
2.7.4

